### DISCUSSION TITLE
Requesting Clarifications for Distributed Setup in NAT'd Environment

### DISCUSSION THREAD

**Discussion by bicfordio** [Security-Onion-Solutions/securityonion#4401] [Unsupported Versions] [answered]
I have read a lot of different posts about issues with NAT'd environments and how the Manager node cannot be behind a NAT. Specifically, I saw the comments from TOoSmOoth [here](https://github.com/Security-Onion-Solutions/securityonion/discussions/2931) which states it does not work.

I have a distributed setup running 2.3.30, as it's been the most stable and reliable version for my environment. 
* I have proven out a distributed model with 15 Heavy Nodes that are able to build and sync to a Manager node in my lab environment in which there are no NATs in place. 
* This lab environment has a firewall in place, which was a great exercise to determine the flows that are required for my production environment.
* This lab environment obviously has no issues and I can see everything that I would expect within the Manager GUI (alerts within the alerts/hunt/kibana pages, Grid members, Grafana statistics, etc).

For my production build, I build a distributed setup running 2.3.30 within 5 different Hubs that are outside of my core network. 
* Each hub has 4 Heavy nodes for a total of 20. My manager node is hosted in my core network. 
* There is a firewall within each hub that has multiple virtual systems, and then also a firewall in place within the core network.
* There is a NAT IP applied to the manager node for inbound traffic, and another, different NAT applied to the manager for outbound traffic. I don't really have any control over this, as it is just how my enterprise has their network architecture defined.
* Each of the heavy nodes have their own NAT as well. 

After reading through many of the threads here, I realize this NAT complexity in my  is going to provide plenty of issues. But with that said, I am curious as to the following:
* If the manager cannot be behind a NAT, why are the Heavy Nodes able to sync to it and build appropriately?
* I can see the stats of each of the Heavy Nodes within Grafana
* The manager lists all of the accepted keys for each of the Heavy nodes
* All traffic from each of the firewalls involved is completing successfully from node to manager and vice versa
* I can do sudo salt \* test.ping successfully and all nodes respond
* There are no errors within sudo salt-call state.highstate from either node
* The only issue I can really see within the Kibana.log is that there is an econnrefused alert to the manager over port 9200

Would appreciate any suggestions or input in this situation.

**Comment by bicfordio** [comment]
Apologies, as I left out an important detail - I should add that the manager node in my production setup is not receiving any logs within the alerts/hunt/kibana page, and none of the nodes are listed within the Grid tab.

**Comment by TOoSmOotH** [comment, marked as answer]
With all of this NAT I would suggest an overlay VPN network. Here are the main issues around NAT:

- The sensor sends its local IP address to the manager to be opened by the firewall. As long as you manually add the NAT address to the firewall then things should work.
- The Manager uses strict SSL so that the minions know it's legit before sending data to it. The certificate generated on the manager points again to its local IP. This is why it cannot be a NAT address. The sensor will look at the NAT IP and say this doesn't match the certificate so I don't trust you. 
- Heavy nodes require the manager to connect directly to the heavy node for searches. Since it only has its local address it can't find its way there. You would need to revamp the cross cluster config to include the proper IP addresses. This can technically be done in the data pillar but this is not something we have really tested. Another issue here is when we flick the switch on strict SSL for Elastic, the certs will get rejected since the IP in the cert is that of the local IP and note the remote. Each node has an SSL cert that is unique to that node.

**Reply by bicfordio** [reply]
Thank you for explaining points 2 and 3 a bit further. I did add the local IP address to the manager and like you mention, things work accordingly. But it makes total sense that the certs will get rejected because of the NAT'd IP and that is why I'm not actually able to see any sort of alerting data.
